### ISSUE TITLE
Bug: React roots are announced as clickable to screen readers [ Please reopen Issue #20895 ]

### ISSUE THREAD

**Issue by a11y1** [facebook/react#34526]
React version: 17.0.1

Steps To Reproduce
Activate your screen reader
I used [NVDA](https://www.nvaccess.org/), but [JAWS](https://www.freedomscientific.com/products/software/jaws/) would also work
This will not work with VoiceOver because it does not have the same heuristic about elements with click handlers
Go to a page that has one or more React 17 root nodes. For example: https://qc9xj.csb.app/
Use the regular navigation of your screen reader to explore the page
In the case of NVDA, this would be using Arrow keys up and down
Hear your screen reader announcing the H2 as "clickable"
Link to code example: https://codesandbox.io/s/epic-babbage-qc9xj (https://qc9xj.csb.app/)

The current behavior
Here is the log of what NVDA says as I'm navigating the page:

content info landmark heading level 1 This is a static HTML footer
(I press Arrow Down)
clickable heading level 2 This is a heading within a React application.
(I press Arrow Down)
banner landmark heading level 1 This is a static HTML header

The expected behavior
It should not announce the heading as clickable/interactive. Here is what it would sound like without React on the page:

content info landmark heading level 1 This is a static HTML footer
(I press Arrow Down)
heading level 2 This is a heading within a React application.
(I press Arrow Down)
banner landmark heading level 1 This is a static HTML header

It should be noted that this is how it used to work in React 16 because the delegation of events was done on the document node, not the React root node.

Notes
Some screen readers like JAWS and NVDA use the presence of a click handler on a node as an indicator that they are interactive
Because of React's event delegation, all synthetic events are handled at one central node. In React 16 it's the document and in React 17 it's the root node.
The example shows that it's announced as clickable even when there is no interactive content inside. But even if the React node contained interactive content, the parent container should not be announced as interactive - because it is not.
This is also called out by the Accessibility Inspector in the Firefox DevTools:
"Does not meet WCAG standards for accessibility. Clickable elements must be focusable and should have interactive semantics. [Learn more](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)"
While this is already an issue with just one React root on the page, this problem will get increasingly worse and annoying for the user when there are multiple React roots on the page
This could easily happen on pages that are constructed with micro frontend frameworks or widgets

**Comment by eps1lon** [comment]
Closing in favor of https://github.com/facebook/react/issues/20895
